Integrations/McAfee_DXL/McAfee_DXL.py

```diff
@@ -0,0 +1,163 @@
+            text_file.write(demisto.params().get('broker_ca_bundle'))
+        with open(self.cert_file, "w") as text_file:
+            text_file.write(demisto.params().get('cert_file'))
+        with open(self.private_key, "w") as text_file:
+            text_file.write(demisto.params().get('private_key'))
+
+        self.broker_urls = demisto.params().get('broker_urls').split(',')
+        self.push_ip_topic = demisto.params().get('push_ip_topic')
+        self.push_url_topic = demisto.params().get('push_url_topic')
+        self.push_domain_topic = demisto.params().get('push_domain_topic')
+        self.push_hash_topic = demisto.params().get('push_hash_topic')
+
+    def push_ip(self, ip, trust_level):
+        if not (is_ip_valid(ip)):
+            return create_error_entry(f'argument ip {ip} is not a valid IP')
+
+        try:
+            trust_level_key = get_trust_level_key(trust_level)
+            self.send_event(self.push_ip_topic, f'ip:{ip};trust_level:{trust_level_key}')
+            return f'Successfully pushed ip {ip} with trust level {trust_level}'
+        except Exception as exception:
+            return create_error_entry(str(exception))
+
+    def push_url(self, url, trust_level):
+        try:
+            trust_level_key = get_trust_level_key(trust_level)
+            self.send_event(self.push_url_topic, f'url:{url};trust_level:{trust_level_key}')
+            return f'Successfully pushed url {url} with trust level {trust_level}'
+        except Exception as exception:
+            return create_error_entry(str(exception))
+
+    def push_domain(self, domain, trust_level):
+        try:
+            trust_level_key = get_trust_level_key(trust_level)
+            self.send_event(self.push_domain_topic, f'domain:{domain};trust_level:{trust_level_key}')
+            return f'Successfully pushed domain {domain} with trust level {trust_level}'
+        except Exception as exception:
+            return create_error_entry(str(exception))
+
+    def push_hash(self, hash_obj, trust_level):
+        try:
+            trust_level_key = get_trust_level_key(trust_level)
+            self.send_event(self.push_hash_topic, f'hash:{hash_obj};trust_level:{trust_level_key}')
+            return f'Successfully pushed hash {hash_obj} with trust level {trust_level}'
+        except Exception as exception:
+            return create_error_entry(str(exception))
+
+    def get_client_config(self):
+        config = DxlClientConfig(
+            broker_ca_bundle=self.broker_ca_bundle,
+            cert_file=self.cert_file,
+            private_key=self.private_key,
+            brokers=[Broker.parse(url) for url in self.broker_urls]
+        )
+        config.connect_retries = 1
+        config.reconnect_delay = 1
+        config.reconnect_delay_max = 10
+        return config
+
+    def validate(self):
+        config = self.get_client_config()
+        with DxlClient(config) as client:
+            client.connect()
```

Comment:
i think it would be better if we `connect` / `disconnect` the client in one flow and not in each function